iceoryx_utils/include/iceoryx_utils/internal/cxx/unique_ptr.inl

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_UTILS_CXX_UNIQUE_PTR_INL
+#define IOX_UTILS_CXX_UNIQUE_PTR_INL
+
+namespace iox
+{
+namespace cxx
+{
+template <typename T>
+unique_ptr<T>::unique_ptr(function_ref<void(ptr_t)>&& deleter) noexcept
+    : m_deleter(deleter)
+{
+}
+
+template <typename T>
+unique_ptr<T>::unique_ptr(ptr_t ptr, function_ref<void(ptr_t)>&& deleter) noexcept
+    : m_ptr(ptr)
+    , m_deleter(std::move(deleter))
+{
+}
+
+template <typename T>
+unique_ptr<T>::unique_ptr(std::nullptr_t) noexcept
+{
```

Comment:
Please try to use always constructor delegation. You have on constructor which provides access for the whole construction process and the other constructors delegate their stuff to this constructor. In the `unique_ptr` case it would be the ctor: `unique_ptr<T>::unique_ptr(ptr_t ptr, function_ref<void(ptr_t)>&& deleter) noexcept`
and every other ctor should delegate to it.
Then you would realize that this `unique_ptr` ctor has a problem since it does not define a deleter.